### PR TITLE
Read all the xml data and create collections

### LINKED ISSUE #26: Use the new cli_utils feature to get rid of remove_actions
See: https://github.com/Kinto/kinto.py/pull/63

### PR DESCRIPTION
**Pull request by Natim**
r? @almet @leplatrem @magopian 

Also fixes #26

### CODE REVIEW

File: schemas/addons.json [hunk 1]
```diff
@@ -0,0 +1,129 @@
+{
+  "title": "Addon",
+  "description": "Addon",
```
Comment (magopian) [inline schemas/addons.json]: Maybe something more explicit like we have [here](https://github.com/mozilla-services/xml2kinto/pull/25/files#diff-9431e21f44d87c781deb4ab6f5979fc4R3)? `An addon blocklist entry`?

File: xml2kinto/__main__.py [hunk 2]
```diff
@@ -1,17 +1,92 @@
...
-collection_name = u'certificates'
+COLLECTION_PERMISSIONS = {'read': ["system.Everyone"]}
+cert_bucket = u'blocklists'
+cert_collection = u'certificates'
+gfx_bucket = u'blocklists'
+gfx_collection = u'gfx'
+addons_bucket = u'blocklists'
+addons_collection = u'addons'
+plugins_bucket = u'blocklists'
+plugins_collection = u'plugins'
```
Comment (leplatrem) [inline xml2kinto/__main__.py]: nit: why those aren't upercase?

Comment (magopian) [inline xml2kinto/__main__.py]: They will be, they need to be, thanks for spotting it ;)

File: xml2kinto/__main__.py [hunk 3]
```diff
@@ -1,17 +1,92 @@
...
+        xpath=xpath)
+    kinto_records = get_kinto_records(
+        kinto_client=kinto_client,
+        bucket=bucket,
+        collection=collection,
+        permissions=COLLECTION_PERMISSIONS)
+
+    diff = get_diff(xml_records, kinto_records)
+    synchronize(diff, kinto_client,
+                bucket=bucket, collection=collection)
```
Comment (leplatrem) [inline xml2kinto/__main__.py]: If `synchronize` is not in charge of building the `diff`, maybe it should be named `push records` ?

File: xml2kinto/id_generator.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import hashlib
+import uuid
+from six import text_type
+
+
+def create_id(data):
```
Comment (leplatrem) [inline xml2kinto/id_generator.py]: The code of this method deserves some comments :)

File: xml2kinto/id_generator.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+                else:
+                    hash.update(item.encode('utf-8'))
+        elif isinstance(value, dict):
+            keys = sorted(value.keys())
+            for key in keys:
+                new_value = '%s:%s' % (key, value[key])
+                hash.update(new_value.encode('utf-8'))
+        else:
+            hash.update(value.encode('utf-8'))
+    return text_type(uuid.UUID(hash.hexdigest()))
```
Comment (magopian) [inline xml2kinto/id_generator.py]: I might be missing something, but is this going to work properly for deeply nested structures? Don't we need to run this recursively on the data?

File: xml2kinto/synchronize.py [hunk 6]
```diff
@@ -1,53 +1,39 @@
...
+def synchronize(diff, kinto_client, bucket, collection):
+    to_create, to_delete = diff
+
+    print('Syncing to {}{}'.format(
+        kinto_client.session_kwargs['server_url'],
+        kinto_client.endpoints.get(
+            'records', bucket=bucket, collection=collection)))
+
+    print('- {} records to create.'.format(len(to_create)))
+    print('- {} records to delete.'.format(len(to_delete)))
```
Comment (leplatrem) [inline xml2kinto/synchronize.py]: Maybe use `logging` since we can have `--verbose` on CLI

File: xml2kinto/xml.py [hunk 7]
```diff
@@ -0,0 +1,92 @@
+from __future__ import absolute_import
+from xml.etree import ElementTree
+
+from .id_generator import create_id
+
+XML_NAMESPACE = '{http://www.mozilla.org/2006/addons-blocklist}'
+SUPPORTED_OPTIONS = ['name', 'xpath', 'fields']
+
+
+def get_info(fields, data):
```
Comment (leplatrem) [inline xml2kinto/xml.py]: Comment ? 

`get_info` -> `get_record` ?

File: xml2kinto/id_generator.py (hunk 5 above)
Comment (Natim) [inline xml2kinto/id_generator.py]: We didn't had the need for recursiveness yet but that is a good point.

File: .travis.yml [hunk 8]
```diff
@@ -3,11 +3,17 @@ python: 2.7
 env:
     - TOX_ENV=py27
     - TOX_ENV=py34
+    #- TOX_ENV=py35
```
Comment (Natim) [inline .travis.yml]: We should remove that since it is now defined in the matrix below.

File: schemas/addons.json [hunk 9]
```diff
@@ -0,0 +1,129 @@
+{
```
Comment (Natim) [inline schemas/addons.json]: This are old versions of the schema and should be removed.

File: schemas/gfx.json [hunk 10]
```diff
@@ -0,0 +1,77 @@
+{
```
Comment (Natim) [inline schemas/gfx.json]: This are old versions of the schema and should be removed.

File: schemas/plugins.json [hunk 11]
```diff
@@ -0,0 +1,123 @@
+{
```
Comment (Natim) [inline schemas/plugins.json]: This are old versions of the schema and should be removed.

**Comment by Natim** [comment]
I am landing this.

**Pull request merged**